src/wp-includes/load.php

```diff
@@ -335,6 +337,20 @@ function wp_in_development_mode( $mode ) {
 		return true;
 	}
 
+	$valid_modes = array(
+		'core',
+		'plugin',
+		'theme',
+		'all',
+	);
+
+	/**
+	 * If no mode is passed, return true if any valid development mode is set.
+	 */
+	if ( empty( $mode ) && in_array( $current_mode, $valid_modes, true ) ) {
+		return true;
+	}
```

Comment:
See https://github.com/WordPress/wordpress-develop/pull/4850#discussion_r1265435770, this is a functional change, so it clearly goes beyond what we discussed before regarding naming.